File: RA1/clase10/JAVA/Gato.java
public class Gato{
    //Atributos
    private String name;
    private String sex;
    int age;
    int weight;
    private String color;
    private String texture;

    //Metodos
    //El primer metodo es el Constructor
    //Permite instanciar a la clase, es decir, entrega la estructura y los metodos al objeto creado.
    public Gato(String name, String sex, int age, int weight, String color, String texture){
        this.name = name;
        this.sex = sex;
        this.age = age;
        this.weight = weight;
        this.color = color;
        this.texture = texture;
    }

    public void eat(){
        System.out.println(this.name + " esta comiendo...");
    }
    public void move(){
        System.out.println(this.name + " esta caminando...");
    }

    public void meow(){
        System.out.println(this.name + " esta maullando...");
    }
    public void purr(){
        System.out.println(this.name + " esta ronrroneando...");
    }

    public void huntMice(){
        System.out.println(this.name + " esta cazando ratonesb...");
    }

    //Getters
    // Son metodos que permiten obtener informacion del objeto

    public String getName(){
        return this.name;
    }

    public String getSex(){
        return this.sex;
    }

    public String getColor(){
        return this.color;
    }

    public String getTexture(){
        return this.texture;
    }


    //Setters
    // Son metodos que permiten modificar o cambiar valores de los atributos de un objeto

    public void setName(String newName){
        this.name = newName;
    }
    public void setSex(String newSex){
        if(newSex == "macho" || newSex == "hembra")
            this.sex = newSex;
        else
            System.out.println("Cambio denegado...:(");
    }
    public void setColor(String newColor){
        if(newColor == "naranjo" || newColor == "blanco" || newColor == "negro" || newColor == "gris" || newColor == "marron")
            this.color = newColor;
        else
            System.out.println("Cambio denegado... :(");
    }

    public void setTexture(String newTexture){
        if(newTexture == "rayado" || newTexture == "lisa" || newTexture == "pelado")
            this.texture = newTexture;
        else
            System.out.println("Cambio denegado... :(");
    }

}

File: RA1/clase10/PYTHON/Gato.py
class Gato:
    # atributos
    #metodos

    def __init__(self, name, sex, age, weight, color, texture):
        self.__name = name
        self.__sex = sex
        self.age = age
        self.weight = weight
        self.__color = color
        self.__texture = texture

    def eat(self):
        print(self.__name + " esta comiendo...")

    def move(self):
        print(self.__name + " esta caminando...")

    def meow(self):
        print(self.__name + " esta maullando...")

    def purr(self):
        print(self.__name + " esta ronrroneando...")

    def hunt_mice(self):
        print(self.__name + " esta cazando ratones...")
    
    # Getter en Python

    def get_name(self):
        return self.__name
    
    def get_sex(self):
        return self.__sex

    def get_color(self):
        return self.__color

    def get_texture(self):
        return self.__texture

    
    #Setters

    def set_name(self, new_name):
        self.__name = new_name
    
    def set_sex(self, new_sex):
        if new_sex == "macho" or new_sex == "hembra":
            self.__sex = new_sex
        else:
            print("Cambio denegado.. :(")

    def set_color(self, new_color):
        if new_color in ["naranjo", "blanco", "negro", "gris", "marron"]:
            self.__color = new_color
        else:
            print("Cambio denegado.. :(")
            
    def set_texture(self, new_texture):
        if new_texture in ["rayado", "lisa", "pelado"]:
            self.__texture = new_texture
        else:
            print("Cambio denegado.. :(")


    